lib/membrane/video_compositor/scene.ex

```diff
@@ -0,0 +1,161 @@
+defmodule Membrane.VideoCompositor.Scene do
+  @moduledoc """
+  Scene describes positions and transformations of the videos, provided to the video compositor.
+  """
+
+  # alias Membrane.VideoCompositor.Position
+  alias Membrane.VideoCompositor.Scene.ComponentsManager, as: Manager
+  alias Membrane.VideoCompositor.Scene.Element
+  alias Membrane.VideoCompositor.Scene.ElementDescription
+  alias Membrane.VideoCompositor.Scene.Video
+
+  @type error_t :: any()
+  @type id_t :: non_neg_integer()
+  @type component_t :: module()
+  @type components_t :: keyword(component_t())
+
+  @typedoc """
+  Description of the scene.
+  """
+  @type scene_description_t() :: [
+          videos: %{required(id_t()) => ElementDescription.t()},
+          scenes: %{required(atom()) => t()}
+        ]
+
+  @typedoc """
+  Internal state of the scene.
+  """
+  @type t :: %__MODULE__{
+          videos: %{required(id_t()) => Video.t()},
+          components: components_t(),
+          scenes: %{required(atom()) => t()},
+          state: %{}
+        }
+  defstruct videos: %{}, components: [], scenes: %{}, state: %{}
+
+  @spec init(scene_description_t(), Manager.t()) ::
+          {:ok, {t(), Manager.t()}} | {:error, error_t()}
+  def init(scene_description, manager) do
+    videos = Keyword.get(scene_description, :videos, %{})
+    scenes = Keyword.get(scene_description, :scenes, %{})
+    scene_description = Keyword.drop(scene_description, [:videos, :scenes])
+
+    scene = %__MODULE__{}
+
+    with {:ok, _size} <- Keyword.fetch(scene_description, :size),
+         {:ok, {scene, manager}} <- add_videos(scene, manager, videos),
+         {:ok, {scene, manager}} <- add_scenes(scene, manager, scenes),
+         element_description = ElementDescription.init(scene_description),
+         {:ok, {manager, state}} <- Manager.register_element(manager, element_description) do
+      components = ElementDescription.get_components(element_description)
+      scene = %__MODULE__{scene | state: state, components: components}
+
+      {:ok, {scene, manager}}
+    else
+      :error -> {:error, "Scene must have `:size` property specified"}
+      {:error, error} -> {:error, error}
+    end
```

Comment:
I'm not found of that style of handling errors, but I think I can understand reasons why you wrote it like that. Maybe we could discuss that matter with the rest of the them, perhaps on the architecture brainstorming meeting?